### PR TITLE
feat(common): ldml: import error messages from the compiler   🙀

### PR DESCRIPTION
**Pull request by srl295**
For: #7534

- move the import related Errors into messages
- comprehensive test cases for import/implied

@keymanapp-test-bot skip

### CODE REVIEW

File: common/web/types/test/helpers/reader-callback-test.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
+import 'mocha';
+import {assert} from 'chai';
+import {loadLdmlKeyboardSchema, loadFile, makePathToFixture} from '../helpers/index.js';
+import LDMLKeyboardXMLSourceFileReader from '../../src/ldml-keyboard/ldml-keyboard-xml-reader.js';
+import { CompilerCallbacks, CompilerEvent } from '../../src/util/compiler-interfaces.js';
+import { LDMLKeyboardXMLSourceFile } from '../../src/ldml-keyboard/ldml-keyboard-xml.js';
+
+// TODO-LDML: this is largely a port from developer/src/kmc-keyboard/test/helpers/index.ts
```
Comment (srl295) [inline common/web/types/test/helpers/reader-callback-test.ts:8]: Stop me if you've seen this before…

these two are slightly different.  Might share some common code.

File: common/web/types/src/ldml-keyboard/ldml-keyboard-xml-reader.ts [hunk 2]
```diff
@@ -102,46 +102,53 @@ export default class LDMLKeyboardXMLSourceFileReader {
...
   }
 
   private resolveOneImport(obj: any, subtag: string, asImport: LKImport) {
-    if (asImport.base !== constants.cldr_import_base) {
-      // TODO-LDML: use custom error class CompilerImportResolveError
-      throw new Error(`import element with base ${asImport.base} is unsupported, only ${constants.cldr_import_base} is supported.`);
+    const { base, path } = asImport;
+    if (base !== constants.cldr_import_base) {
+      this.callbacks.reportMessage(CommonTypesMessages.Error_ImportInvalidBase({base, path, subtag}));
+      return;
```
Comment (mcdurdin) [inline common/web/types/src/ldml-keyboard/ldml-keyboard-xml-reader.ts:124]: We should probably have a `return false;` for failure cases?

File: common/web/types/src/util/common-events.ts [hunk 3]
```diff
@@ -1,21 +1,47 @@
...
   // some other unknown Ajv error. Same as above, but without structure
-  static Fatal_UnknownXMLValidationError = (o:{errorsText: string}) => m(this.FATAL_XMLValidationError,
+  static Fatal_UnknownXMLValidationError = (o: { errorsText: string }) => m(this.FATAL_XMLValidationError,
     `Validation failed: ${o.errorsText}`);
   static FATAL_XMLValidationError = SevFatal | 0x0001;
+
+  static Error_ImportInvalidBase = (o: { base: string, path: string, subtag: string }) =>
+    m(this.ERROR_ImportInvalidBase,
+      `Import element with base ${o.base} is unsupported. Only ${constants.cldr_import_base} is supported.`);
+  static ERROR_ImportInvalidBase = SevError | 0x0001;
```
Comment (mcdurdin) [inline common/web/types/src/util/common-events.ts:26]: Each identifier should be unique, XOR its severity mask. We have had messages change in severity in the past, and this keeps things sane.

File: common/web/types/test/helpers/reader-callback-test.ts (hunk 1 above)
Comment (mcdurdin) [reply common/web/types/test/helpers/reader-callback-test.ts:8]: Let's look at refactoring that as a possible future lowpri task

**Review by mcdurdin — APPROVED** [review]
LGTM, although adjustments will need to be made on basis of #8133 review.

File: common/web/types/test/helpers/reader-callback-test.ts (hunk 1 above)
Comment (srl295) [reply common/web/types/test/helpers/reader-callback-test.ts:8]: it's already TODO-LDML.

File: common/web/types/src/util/common-events.ts (hunk 3 above)
Comment (srl295) [reply common/web/types/src/util/common-events.ts:26]: moot now that they are all errors, but good to know

**Comment by srl295** [comment]
@mcdurdin  updated per comments and rebased.

File: common/web/types/src/ldml-keyboard/ldml-keyboard-xml-reader.ts (hunk 2 above)
Comment (srl295) [reply common/web/types/src/ldml-keyboard/ldml-keyboard-xml-reader.ts]: done

**Comment by srl295** [comment]
@mcdurdin  this was already approved. but it's ready for re-merge

**Review by mcdurdin — APPROVED** [review]

**Comment by srl295** [comment]
web failures seem unrelated.

**Pull request merged**